weave/trace_server/file_storage.py

```diff
@@ -211,6 +242,9 @@ def _get_client(self, account: str) -> BlobServiceClient:
     @create_retry_decorator("azure_storage")
     def store(self, uri: AzureFileStorageURI, data: bytes) -> None:
         """Store data in Azure container with automatic retries on failure."""
+        assert isinstance(uri, AzureFileStorageURI) and uri.to_uri_str().startswith(
```

Comment:
why do we have to assert the uri bit?

Reply:
because we aren't guaranteed these uri's are for the given type, surprised we don't just rely on the type hints from the parse fn but I get it

Reply:
I want to be really careful here in the case that a url stored points to storage X, but then the client is storage Y and we blindly use the path of X.